nucliadb/nucliadb/ingest/service/writer.py

```diff
@@ -502,25 +533,54 @@ async def SetEntities(self, request: SetEntitiesRequest, context=None) -> OpStat
             await txn.commit(resource=False)
         return response
 
+    async def UpdateEntitiesGroup(  # type: ignore
+        self, request: UpdateEntitiesGroupRequest, context=None
+    ) -> UpdateEntitiesGroupResponse:
+        response = UpdateEntitiesGroupResponse()
+        async with self.proc.driver.transaction() as txn:
+            kbobj = await self.proc.get_kb_obj(txn, request.kb)
+            if kbobj is None:
+                response.status = UpdateEntitiesGroupResponse.Status.KB_NOT_FOUND
+                return response
+
+            entities_manager = EntitiesManager(kbobj, txn)
+
+            try:
+                updates = {**request.add, **request.update}
+                await entities_manager.update_entities(request.group, updates)
+                await entities_manager.delete_entities(request.group, request.delete)  # type: ignore
+            except KeyError:
+                response.status = (
+                    UpdateEntitiesGroupResponse.Status.ENTITIES_GROUP_NOT_FOUND
+                )
+                return response
+
+            await txn.commit(resource=False)
+            response.status = UpdateEntitiesGroupResponse.Status.OK
+            return response
+
     async def DelEntities(self, request: DelEntitiesRequest, context=None) -> OpStatusWriter:  # type: ignore
-        txn = await self.proc.driver.begin()
-        kbobj = await self.proc.get_kb_obj(txn, request.kb)
         response = OpStatusWriter()
-        if kbobj is not None:
+        async with self.proc.driver.transaction() as txn:
+            kbobj = await self.proc.get_kb_obj(txn, request.kb)
+            if kbobj is None:
+                await txn.abort()
```

Comment:
NIT: no need to abort if you are using the transaction context manager. It's still ok to be explicit though